core/internal/api/send.go

```diff
@@ -21,10 +22,37 @@ func (client *clientImpl) Send(req *Request) (*http.Response, error) {
 	for headerKey, headerValue := range req.Headers {
 		retryableReq.Header.Set(headerKey, headerValue)
 	}
-	client.setClientHeaders(retryableReq)
-	client.setAuthHeaders(retryableReq)
 
-	resp, err := client.retryableHTTP.Do(retryableReq)
+	return client.sendToWandbBackend(retryableReq)
+}
+
+func (client *clientImpl) Do(req *http.Request) (*http.Response, error) {
+	retryableReq, err := retryablehttp.FromRequest(req)
+	if err != nil {
+		return nil, fmt.Errorf("api: failed to parse request: %v", err)
+	}
+
+	if req.URL.Host == client.backend.baseURL.Host &&
+		strings.HasPrefix(req.URL.Path, client.backend.baseURL.Path) {
+
+		return client.sendToWandbBackend(retryableReq)
+	}
+
+	return client.sendArbitrary(retryableReq)
+}
+
+func (client *clientImpl) sendToWandbBackend(
```

Comment:
i think we can remove this function and just have the logic in the Do function, i don't think it adds for readability

Reply:
hmm the issue is that I use this in two places: `Do` and `Send`. I wouldn't want to duplicate setting the headers in both spots.